contracts/YieldCrabLever.sol

```diff
@@ -0,0 +1,451 @@
+     * @dev _ethToDeposit must be less than msg.value plus the proceeds from the flash swap
+     * @dev the difference between _ethToDeposit and msg.value provides the minimum that a user can receive for their sold wSqueeth
+     * @param _ethToDeposit total ETH that will be deposited in to the strategy which is a combination of msg.value and flash swap proceeds
+     * @param _poolFee Uniswap pool fee
+     */
+    function flashDeposit(uint256 _ethToDeposit, uint24 _poolFee)
+        external
+        payable;
+
+    /**
+     * @notice flash withdraw from strategy, providing strategy tokens, buying wSqueeth, burning and receiving ETH
+     * @dev this function will execute a flash swap where it receives wSqueeth, burns, withdraws ETH and then repays the flash swap with ETH
+     * @param _crabAmount strategy token amount to burn
+     * @param _maxEthToPay maximum ETH to pay to buy back the wSqueeth debt
+     * @param _poolFee Uniswap pool fee
+
+     */
+    function flashWithdraw(
+        uint256 _crabAmount,
+        uint256 _maxEthToPay,
+        uint24 _poolFee
+    ) external;
+}
+
+/// @title A contract to help users build levered position on crab strategy token
+///        using ETH/WETH/DAI/USDC as collateral
+/// @author iamsahu
+/// @notice Each external function has the details on how this works
+contract YieldCrabLever is YieldLeverBase {
+    using TransferHelper for IERC20;
+    using TransferHelper for IMaturingToken;
+    using CastU128I128 for uint128;
+    using CastU256U128 for uint256;
+
+    ICrabStrategy public constant crabStrategy =
+        ICrabStrategy(0x3B960E47784150F5a63777201ee2B15253D713e8);
+
+    ISwapRouter public constant swapRouter =
+        ISwapRouter(0xE592427A0AEce92De3Edee1F18E0157C05861564);
+    IERC20 public crab;
+    bytes6 public constant crabId = 0x333800000000;
+    bytes6 public constant wethId = 0x303000000000;
+    bytes6 public constant daiId = 0x303100000000;
+    bytes6 public constant usdcId = 0x303200000000;
+
+    constructor(Giver giver_) YieldLeverBase(giver_) {}
+
+    /// @notice Invest by creating a levered vault.
+    /// The steps are as follows:
+    /// 1. Based on the ilk transfer it from the user
+    /// @param seriesId The series to invest in
+    /// @param ilkId The collateral to use
+    /// @param amountToInvest The amount of ETH/WETH/DAI/USDC supplied by the user
+    /// @param borrowAmount The amount of fyToken to be borrowed
+    /// @param minCollateral to be received
+    function invest(
+        bytes6 seriesId,
+        bytes6 ilkId,
+        uint256 amountToInvest,
+        uint256 borrowAmount,
+        uint256 minCollateral
+    ) external payable returns (bytes12 vaultId) {
+        IPool pool = IPool(ladle.pools(seriesId));
+        IMaturingToken fyToken = pool.fyToken();
+        // Depend on ilkId we will have to choose the operation
+        if (ilkId != wethId) {
```

Comment:
```suggestion
        if (ilkId != wethId || msg.value > 0) {
```
Unless we are using Ether, and we receive Ether, pull from the user. You can remove the `else`.